adafruit_imageload/bmp/__init__.py

```diff
@@ -62,12 +62,20 @@ def load(
     file.seek(0x2E)  # Number of colors in the color palette
     colors = int.from_bytes(file.read(4), "little")
 
-    if colors == 0 and color_depth >= 16:
-        raise NotImplementedError("True color BMP unsupported")
-
     if compression > 2:
         raise NotImplementedError("bitmask compression unsupported")
```

Comment:
Thanks for adding this great functionality @makermelissa!

Do you know if this check on compression level still necessary? 

I was attempting to test this change with 16, 24, and 32bit images that I got by exporting from GIMP and I found that only the 24bit images could be loaded using this version of the library. The 16bit, and the 32bit variations that I tried all raised this error about the compression.

But I noticed with at least some of those images that raise this error if I comment out the if statement here the image does get loaded and displayed properly. 

I also modified it on my device to print the `compression` level that was found, and was able to see that `3` is the value of compression for all of the images I tested. Maybe instead of removing this logic altogether it could be bumped up to 3 instead of 2.

I will do more thorough testing of all of the different types again later with this commented out and report back the results of that here.

Reply:
> Do you know if this check on compression level still necessary?

Yes. It looks like You can find information on the bitmap format at https://en.wikipedia.org/wiki/BMP_file_format. According to that page, a compression level of 3 means that it is using a bit field mask. Here's the table of compression levels:

| Value |   Identified by   |                      Compression method                      |                           Comments                           |
| :---: | :---------------: | :----------------------------------------------------------: | :----------------------------------------------------------: |
|   0   |      BI_RGB       |                             none                             |                         Most common                          |
|   1   |      BI_RLE8      | [RLE](https://en.wikipedia.org/wiki/Run-length_encoding) 8-bit/pixel |          Can be used only with 8-bit/pixel bitmaps           |
|   2   |      BI_RLE4      |                       RLE 4-bit/pixel                        |          Can be used only with 4-bit/pixel bitmaps           |
|   3   |   BI_BITFIELDS    |                OS22XBITMAPHEADER: Huffman 1D                 | BITMAPV2INFOHEADER: RGB bit field masks, BITMAPV3INFOHEADER+: RGBA |
|   4   |      BI_JPEG      |                  OS22XBITMAPHEADER: RLE-24                   | BITMAPV4INFOHEADER+: [JPEG](https://en.wikipedia.org/wiki/JPEG) image for printing[[14\]](https://en.wikipedia.org/wiki/BMP_file_format#cite_note-onlyprint-14) |
|   5   |      BI_PNG       |                                                              | BITMAPV4INFOHEADER+: [PNG](https://en.wikipedia.org/wiki/Portable_Network_Graphics) image for printing[[14\]](https://en.wikipedia.org/wiki/BMP_file_format#cite_note-onlyprint-14) |
|   6   | BI_ALPHABITFIELDS |                     RGBA bit field masks                     | only [Windows CE](https://en.wikipedia.org/wiki/Windows_CE) 5.0 with .NET 4.0 or later |
|  11   |      BI_CMYK      |                             none                             | only [Windows Metafile](https://en.wikipedia.org/wiki/Windows_Metafile) [CMYK](https://en.wikipedia.org/wiki/CMYK_color_model)[[4\]](https://en.wikipedia.org/wiki/BMP_file_format#cite_note-v11-4) |
|  12   |    BI_CMYKRLE8    |                            RLE-8                             | only [Windows Metafile](https://en.wikipedia.org/wiki/Windows_Metafile) CMYK |
|  13   |    BI_CMYKRLE4    |                            RLE-4                             | only [Windows Metafile](https://en.wikipedia.org/wiki/Windows_Metafile) CMYK |

This means that a DIB header needs to be extracted and parsed in addition to the regular BMP header. The DIB header contains a mask about which bits correspond to which fields and could vary (see https://en.wikipedia.org/wiki/BMP_file_format#Pixel_format for more info). If it is loading the bitmaps properly, it might just be getting lucky.

Anyway, I was expecting this PR to be more of a starting point as opposed to something that works for every possible bitmap in these formats.

By default Photoshop does not include these the bitfield info and it looks like you can also do that in Gimp. You'll want to make sure `Do not write color space information` is checked so it doesn't compress and then for 16-bit choose `X1 R5 G5 B5` and for 32-bit choose `X8 R8 G8 B8`, though the other 32-bit option should work as well.
<img width="290" alt="Screenshot 2023-08-06 at 1 43 19 PM" src="https://github.com/adafruit/Adafruit_CircuitPython_ImageLoad/assets/2681417/2b3e7e35-c001-4983-b43a-da5af001bb3a">